src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/EnumShouldNotHaveDuplicatedValues.cs

```diff
@@ -0,0 +1,67 @@
+                                                                                            s_localizableMessageRuleDuplicatedValue,
+                                                                                            DiagnosticCategory.Design,
+                                                                                            DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                                            isEnabledByDefault: false,
+                                                                                            helpLinkUri: null,
+                                                                                            customTags: WellKnownDiagnosticTags.Telemetry);
+
+        private static readonly LocalizableString s_localizableMessageRuleDuplicatedBitwiseValuePart = new LocalizableResourceString(nameof(MicrosoftCodeQualityAnalyzersResources.EnumShouldNotHaveDuplicatedValuesMessageDuplicatedBitwiseValuePart), MicrosoftCodeQualityAnalyzersResources.ResourceManager, typeof(MicrosoftCodeQualityAnalyzersResources));
+        internal static DiagnosticDescriptor RuleDuplicatedBitwiseValuePart = new DiagnosticDescriptor(RuleId,
+                                                                                                   s_localizableTitle,
+                                                                                                   s_localizableMessageRuleDuplicatedBitwiseValuePart,
+                                                                                                   DiagnosticCategory.Design,
+                                                                                                   DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                                                   isEnabledByDefault: false,
+                                                                                                   helpLinkUri: null,
+                                                                                                   customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public sealed override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(RuleDuplicatedValue, RuleDuplicatedBitwiseValuePart);
+
+        public sealed override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSymbolAction(symbolContext =>
+            {
+                var typeSymbol = (INamedTypeSymbol)symbolContext.Symbol;
+                if (typeSymbol.TypeKind != TypeKind.Enum)
+                {
+                    return;
+                }
+
+                var enumMemberValues = typeSymbol.GetMembers()
+                    .Where(m => m.Kind == SymbolKind.Field)
```

Comment:
I don't know whether that's best for the analyzer to do the kind check before the cast or to do `OfType` directly.